app/Actions/Fortify/TwoFactorChallengeView.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace App\Actions\Fortify;
+
+use App\Models\User;
+use Inertia\Inertia;
+use Laravel\Fortify\Contracts\TwoFactorChallengeViewResponse as TwoFactorChallengeViewContract;
+use LaravelWebauthn\Facades\Webauthn;
+
+class TwoFactorChallengeView implements TwoFactorChallengeViewContract
+{
+    /**
+     * Create an HTTP response that represents the object.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
+    public function toResponse($request)
+    {
+        $userId = $request->session()->get('login.id');
+        $user = User::find($userId);
+
+        $data = [];
+        if ($user !== null) {
+            $data['publicKey'] = Webauthn::prepareAssertion($user);
+        }
+
+        return Inertia::render('Auth/TwoFactorChallenge', $data + [
+            'two_factor' => optional($user)->two_factor_secret && ! is_null(optional($user)->two_factor_confirmed_at),
```

Comment:
This line is super hard to read. It's optional + not is null... We should have a variable here, and the variable defined above, so it's easier to understand.

Reply:
Actually it's a copy/paste code from laravel/fortify, so I wouldn't bother too much